pkg/randomint/randomint.go

```diff
@@ -51,7 +62,29 @@ func Factory(conf model.Masking, seed int64, caches map[string]model.Cache) (mod
 		h := fnv.New64a()
 		h.Write([]byte(conf.Selector.Jsonpath))
 		seed += int64(h.Sum64())
-		return NewMask(conf.Mask.RandomInt.Min, conf.Mask.RandomInt.Max, seed), true, nil
+
+		var seeder model.Seeder
+		if jpath := conf.Seed.Field; jpath != "" {
+			sel := model.NewPathSelector(jpath)
+			seeder = func(context model.Dictionary) (int64, bool, error) {
+				e, ok := sel.Read(context)
+				if !ok {
+					return 0, ok, nil
+				}
+				switch t := e.(type) {
+				case json.Number:
+					v, err := t.Int64()
+					return v, true, err
+				default:
+					return 0, false, nil
+				}
+			}
+		} else {
+			seeder = func(context model.Dictionary) (int64, bool, error) {
+				return 0, false, nil
+			}
+		}
```

Comment:
All of this code should be in the Seeder model and not here in the Mask model. Otherwise, there will be code duplication.